x-pack/solutions/observability/plugins/apm/public/components/app/onboarding/agent_config_instructions.tsx

```diff
@@ -52,23 +70,53 @@ export function AgentConfigInstructions({
         createApiKeyLoading={createApiKeyLoading}
       />
       <EuiSpacer />
-
-      <EuiCodeBlock
-        isCopyable
-        copyAriaLabel={i18n.translate(
-          'xpack.apm.onboarding.agentConfigInstructions.euiCodeBlock.copyAriaLabel',
-          {
-            defaultMessage: 'Copy {variantId} agent configuration code',
-            values: { variantId },
-          }
-        )}
-        language={highlightLang || 'bash'}
-        data-test-subj="commands"
-        lineNumbers={lineNumbers}
-        whiteSpace="pre"
+      <EuiCallOut
+        title={i18n.translate('xpack.apm.onboarding.agentConfigInstructions.callout.title', {
+          defaultMessage: `The command below doesn't show secrets values`,
+        })}
+        color="warning"
+        iconType="warning"
       >
-        {commands}
-      </EuiCodeBlock>
+        <p>
+          {i18n.translate('xpack.apm.onboarding.agentConfigInstructions.callout.body', {
+            defaultMessage: 'Copy to clipboard to get the full command with secrets',
+          })}
+        </p>
+      </EuiCallOut>
+      <EuiSpacer size="s" />
+      <EuiPanel color="subdued" borderRadius="none" hasShadow={false}>
+        <EuiFlexGroup direction="row">
+          <EuiFlexItem grow={1}>
+            <EuiCodeBlock
+              copyAriaLabel={i18n.translate(
+                'xpack.apm.onboarding.agentConfigInstructions.euiCodeBlock.copyAriaLabel',
+                {
+                  defaultMessage: 'Copy {variantId} agent configuration code',
+                  values: { variantId },
+                }
+              )}
+              language={highlightLang || 'bash'}
+              data-test-subj="commands"
+              lineNumbers={lineNumbers}
+              whiteSpace="pre"
+            >
+              {commands}
+            </EuiCodeBlock>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              data-test-subj="apmAgentConfigInstructionsButton"
+              aria-label={i18n.translate('xpack.apm.agentConfigInstructions.button.ariaLabel', {
+                defaultMessage: 'Copy commands',
+              })}
+              iconType="copyClipboard"
+              onClick={() => {
+                copyToClipboard(commandsWithSecrets);
+              }}
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiPanel>
```

Comment:
This is looks exactly like the component on `x-pack/solutions/observability/plugins/apm/public/tutorial/config_agent/agent_config_instructions.tsx` Can we create a shared component and reuse?